src/core/server/core_app/bundle_routes/select_compressed_file.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { extname } from 'path';
+import Accept from 'accept';
+import { open } from './fs';
+
+declare module 'accept' {
+  // @types/accept does not include the `preferences` argument so we override the type to include it
+  export function encodings(encodingHeader?: string, preferences?: string[]): string[];
+}
+
+async function tryToOpenFile(filePath: string) {
+  try {
+    return await open(filePath, 'r');
+  } catch (e) {
+    if (e.code === 'ENOENT') {
+      return undefined;
+    } else {
+      throw e;
+    }
+  }
+}
+
+export async function selectCompressedFile(acceptEncodingHeader: string | undefined, path: string) {
+  let fd: number | undefined;
+  let fileEncoding: 'gzip' | 'br' | undefined;
+  const ext = extname(path);
+
+  const supportedEncodings = Accept.encodings(acceptEncodingHeader, ['br', 'gzip']);
+
+  // do not bother trying to look compressed versions for anything else than js or css files
+  if (ext === '.js' || ext === '.css') {
```

Comment:
I did not unit test this specific file, because we would just be mocking the filesystem APIs and the testing process would be tedious for little value. This is covered both by integration and FTR tests anyway.